prisma/schema.prisma

```diff
@@ -115,16 +119,100 @@ model Area {
+  Score                Score[]
 
   @@unique([area_id, area_name, stageId])
 }
 
-
 model Stage {
   stageNumber Int
-  id          Int    @id @default(autoincrement())
-  name        String @unique
+  id          Int     @id @default(autoincrement())
+  name        String  @unique
   areas       Area[]
+  Score       Score[]
+}
+
+// New Models
+
+model Project {
+  id          Int           @id @default(autoincrement())
+  projectId   String        @unique
+  name        String
+  createdAt   DateTime      @default(now())
+  updatedAt   DateTime      @updatedAt
+  assessments Assessment[]
+  users       UserProject[]
+}
+
+model UserProject {
+  userId    String
+  projectId Int
+  user      User    @relation(fields: [userId], references: [id], onDelete: Cascade)
+  project   Project @relation(fields: [projectId], references: [id], onDelete: Cascade)
+
+  @@id([userId, projectId])
+}
+
+model Assessment {
+  id        Int                  @id @default(autoincrement())
+  projectId Int
+  userId    String
+  createdAt DateTime             @default(now())
+  updatedAt DateTime             @updatedAt
+  project   Project              @relation(fields: [projectId], references: [id], onDelete: Cascade)
+  user      User                 @relation(fields: [userId], references: [id])
+  artefacts AssessmentArtefact[]
+  areas     AssessmentArea[]
+  Score     Score[]
+}
+
+model AssessmentArtefact {
+  id           Int        @id @default(autoincrement())
+  artefactId   Int
+  assessmentId Int
+  checkmark    Boolean
+  comment      String?
+  artefact     Artefact   @relation(fields: [artefactId], references: [id])
+  assessment   Assessment @relation(fields: [assessmentId], references: [id], onDelete: Cascade)
+
+  @@unique([artefactId, assessmentId])
+}
+
+model AssessmentArea {
+  id           Int                      @id @default(autoincrement())
+  areaId       Int
+  assessmentId Int
+  comment      String?
+  score        Float
+  area         Area                     @relation(fields: [areaId], references: [id])
+  assessment   Assessment               @relation(fields: [assessmentId], references: [id], onDelete: Cascade)
+  questions    AssessmentAreaQuestion[]
+
+  @@unique([areaId, assessmentId])
+}
+
+model AssessmentAreaQuestion {
+  id               Int            @id @default(autoincrement())
+  assessmentAreaId Int
+  question         String
+  answer           Int?
+  assessmentArea   AssessmentArea @relation(fields: [assessmentAreaId], references: [id], onDelete: Cascade)
+}
+
+model Score {
+  id           Int        @id @default(autoincrement())
+  assessmentId Int
+  areaId       Int?
+  artefactId   Int?
+  stageId      Int?
+  score        Float
+  assessment   Assessment @relation(fields: [assessmentId], references: [id], onDelete: Cascade)
+  area         Area?      @relation(fields: [areaId], references: [id])
+  artefact     Artefact?  @relation(fields: [artefactId], references: [id])
```

Comment:
In the case of artefact + assessment, how the score would be defined? 
Something like: `1 / number_of_artefacts_in_area` ? And
> Sum scores for all artefacts under a specific area.


Would never be 1 because of the float in some cases. Not sure if it is a problem, but in the case of counting score for all artefacts in the stage it would lose some digits if we care about precision.

Reply:
I was thinking it could be both number_of_artefacts_in_area and number_of_artefacts_in_stage. I don't think we will have a rounding issue though? perhaps we force the score to always be a rounded-down integer.